src/miniscript/astelem.rs

```diff
@@ -345,17 +364,34 @@ where
                 hash160::Hash::from_hex(x).map(Terminal::Hash160)
             }),
             ("true", 0) => Ok(Terminal::True),
-            ("and_v", 2) => expression::binary(top, Terminal::AndV),
+            ("false", 0) => Ok(Terminal::False),
+            ("and_v", 2) => {
+                let expr = expression::binary(top, Terminal::AndV)?;
+                if let Terminal::AndV(_, ref right) = expr {
+                    if let Terminal::True = right.node {
+                        return Err(Error::NonCanonicalTrue);
+                    }
+                }
+                Ok(expr)
+            }
             ("and_b", 2) => expression::binary(top, Terminal::AndB),
-            ("and_or", 3) => Ok(Terminal::AndOr(
+            ("andor", 3) => Ok(Terminal::AndOr(
                 expression::FromTree::from_tree(&top.args[0])?,
-                expression::FromTree::from_tree(&top.args[2])?,
                 expression::FromTree::from_tree(&top.args[1])?,
+                expression::FromTree::from_tree(&top.args[2])?,
```

Comment:
I think we need to add a `and_n` entry here

Reply:
Nice catch! I forgot to add this